prime-router/src/main/kotlin/fhirengine/engine/FHIRDestinationFilter.kt

```diff
@@ -0,0 +1,561 @@
+import gov.cdc.prime.router.azure.observability.event.ReportAcceptedEvent
+import gov.cdc.prime.router.azure.observability.event.ReportRouteEvent
+import gov.cdc.prime.router.fhirengine.translation.hl7.SchemaException
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.CustomContext
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.FhirPathUtils
+import gov.cdc.prime.router.fhirengine.utils.FhirTranscoder
+import gov.cdc.prime.router.report.ReportService
+import org.hl7.fhir.r4.model.Base
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.Observation
+import org.jooq.Field
+import java.time.OffsetDateTime
+
+/**
+ * [metadata] mockable metadata
+ * [settings] mockable settings
+ * [db] mockable database access
+ * [blob] mockable blob storage
+ * [queue] mockable azure queue
+ */
+class FHIRDestinationFilter(
+    metadata: Metadata = Metadata.getInstance(),
+    settings: SettingsProvider = this.settingsProviderSingleton,
+    db: DatabaseAccess = this.databaseAccessSingleton,
+    blob: BlobAccess = BlobAccess(),
+    azureEventService: AzureEventService = AzureEventServiceImpl(),
+    reportService: ReportService = ReportService(),
+) : FHIREngine(metadata, settings, db, blob, azureEventService, reportService) {
+
+    /**
+     * The name of the lookup table to load the shorthand replacement key/value pairs from
+     */
+    private val fhirPathFilterShorthandTableName = "fhirpath_filter_shorthand"
+
+    /**
+     * The name of the column in the shorthand replacement lookup table that will be used as the key.
+     */
+    private val fhirPathFilterShorthandTableKeyColumnName = "variable"
+
+    /**
+     * The name of the column in the shorthand replacement lookup table that will be used as the value.
+     */
+    private val fhirPathFilterShorthandTableValueColumnName = "fhirPath"
+
+    /**
+     * Lookup table `fhirpath_filter_shorthand` containing all the shorthand fhirpath replacements for filtering.
+     */
+    private val shorthandLookupTable by lazy { loadFhirPathShorthandLookupTable() }
+
+    /**
+     * Adds logs for reports that pass through various methods in the FHIRRouter
+     */
+    private var actionLogger: ActionLogger? = null
+
+    /**
+     * Load the fhirpath_filter_shorthand lookup table into a map if it can be found and has the expected columns,
+     * otherwise log warnings and return an empty lookup table with the correct columns. This is valid since having
+     * a populated lookup table is not required to run the universal pipeline routing
+     *
+     * @returns Map containing all the values in the fhirpath_filter_shorthand lookup table. Empty map if the
+     * lookup table was not found, or it does not contain the expected columns. If an empty map is returned, a
+     * warning indicating why will be logged.
+     */
+    internal fun loadFhirPathShorthandLookupTable(): MutableMap<String, String> {
```

Comment:
I think this is identical to what's in FHIRRouter... maybe move to somewhere re-usable rather than duplicating it?